coresimd/arm/cmsis.rs

```diff
@@ -0,0 +1,318 @@
+#![allow(non_snake_case)]
+
+/// Extracted from [CMSIS 5]'s CMSIS/Core/Include/cmsis_gcc.h
+///
+/// [CMSIS 5]: https://github.com/ARM-software/CMSIS_5
+
+/* Core function access */
+
+/// Enable IRQ Interrupts
+///
+/// Enables IRQ interrupts by clearing the I-bit in the CPSR. Can only be
+/// executed in Privileged modes.
+#[inline]
+#[target_feature(enable = "mclass")]
```

Comment:
I am 99% sure that `#[target_feature(enable = "mclass")]` is the wrong thing to do here. Instead the whole module should be guarded by `#[cfg(target_feature = "mclass")]`.

M-class ARM is very different from A/R-class ARM and these intrinsics only make sense when targeting a M-class architecture. Keep in mind that code compiled for M-class ARM is not binary compatible with A/R-class ARM, so "enabling" the `mclass` feature does not make any sense.

Reply:
> Instead the whole module should be guarded by #[cfg(target_feature = "mclass")].

That is already the case, and you are correct, this isn't technically necessary. 

We had issues in the past with the whole binary being compiled with a feature enabled not being 100% equivalent with all the functions in the binary having the feature enabled. These annotations should be redundant if everything works fine, view them as just being extra explicit here.